### PR TITLE
Update drawNameTag function and file used this function

### PR DESCRIPTION
**Pull request by kexinF**
@blickly Hi Ben, I saw your comment regarding the rebase of the drawNametag API. Because that main dev is relatively old, I created a new PR based on your request. 

In this PR, we revise the drawNametag function to take only 2 arguments: nametag and handWave.

By clearly defining what parameters are related to nametag or handWave, we expect to increase the readability of the function.

### CODE REVIEW

**Comment by blickly** [comment]
Creating a new PR sounds good, but in that case, let's close https://github.com/aimpowered/ZoomCompanion/pull/24

**Review by blickly — CHANGES_REQUESTED** [review]

File: app/main/NameTag.tsx [hunk 1]
```diff
@@ -3,7 +3,8 @@
 import React, { useEffect, useState } from 'react';
 import Switch from '@mui/material/Switch';
 import { alpha, styled } from '@mui/material/styles';
-import drawNametag from "@/lib/drawNametag";
+// import drawNametag from "@/lib/drawNametag";
```
[the remark this answers is not in the document]
Comment (kexinF) [reply app/main/NameTag.tsx]: Solved

File: lib/drawNametag.ts [hunk 2]
```diff
@@ -1,14 +1,27 @@
...
+  visible: boolean;
+  fullName: string;
+  preferredName: string;
+  pronouns: string;
+  disclosure: string;
+}
+export interface HandWaveBadge {
+  selectedWaveHand: number | null;
+  waveHands: string[];
+}
```
[the remark this answers is not in the document]
Comment (kexinF) [reply lib/drawNametag.ts]: @blickly Yes, I have modified the code based on this structure. But I set waveText can also be null, this is only set to null when visible is null.

File: lib/drawNametag.ts [hunk 3]
```diff
@@ -1,14 +1,26 @@
...
+  preferredName: string;
+  pronouns: string;
+  disclosure: string;
+}
+export interface HandWaveBadge {
+  visible: boolean;
+  waveText: string | null;
+}
+
+// export default function drawNametag(nameTagStatus:boolean, currentNameTag:string[], selectedWaveHand: number | null, waveHands: string[]): ImageData {
```
Comment (blickly) [inline lib/drawNametag.ts:15]: Let's remove the commented out code

File: lib/drawNametag.ts [hunk 4]
```diff
@@ -1,14 +1,26 @@
...
+  visible: boolean;
+  fullName: string;
+  preferredName: string;
+  pronouns: string;
+  disclosure: string;
+}
+export interface HandWaveBadge {
+  visible: boolean;
+  waveText: string | null;
+}
```
Comment (blickly) [inline lib/drawNametag.ts:13]: You can actually include the fact that disabled badges don't need text in the types themselves, like:
```suggestion
interface DisabledBadge {
  visible: false
}
interface EnabledHandWaveBadge {
  visible: true;
  waveText: string;
}
export type HandWaveBadge = DisabledBadge | EnabledHandWaveBadge;
```

File: app/main/NameTag.tsx [hunk 5]
```diff
@@ -40,7 +40,22 @@ function NameTag({
...
+        fullName: inputValues[0],
+        preferredName: inputValues[1],
+        pronouns: inputValues[2],
+        disclosure: inputValues[3],
+    };
+
+    const handWave: HandWaveBadge = {
+        visible: selectedWaveHand !== null ? true : false,
+        waveText: selectedWaveHand !== null ? waveHands[selectedWaveHand] : null,
+    };
```
Comment (blickly) [inline app/main/NameTag.tsx:55]: I find it a little clearer to not repeat the `selectedWaveHand !== null` check twice.  Maybe something like:
```suggestion
    const handWave: HandWaveBadge =
       selectedWaveHand !== null ?
           {visible: true, waveText: waveHands[selectedWaveHand]} :
           {visible: false};
```

If you agree, let's make the same change in page.tsx

File: lib/drawNametag.ts [hunk 6]
```diff
@@ -28,36 +40,34 @@ export default function drawNametag(nameTagStatus:boolean, currentNameTag:string
...
+    context.fillText(nametag.pronouns, 800, 600 + 1 * 50);
 
     context.font = '40px Arial';
-    context.fillText(currentNameTag[3], 800, 600 + 2 * 50);
+    context.fillText(nametag.disclosure, 800, 600 + 2 * 50);
   }
 
 
-  if (selectedWaveHand !== null) {
+  if (handWave.visible && handWave.waveText) {
```
Comment (blickly) [inline lib/drawNametag.ts:56]:
```suggestion
  if (handWave.visible) {
```

**Review by blickly — CHANGES_REQUESTED** [review]

File: lib/drawNametag.ts (hunk 3 above)
Comment (kexinF) [reply lib/drawNametag.ts]: solved

**Review by blickly — APPROVED** [review]
LGTM

**Pull request merged**